traits/observers/expression.py

```diff
@@ -76,11 +85,82 @@ def then(self, expression):
         """
         return SeriesExpression(self, expression)
 
+    def dict_items(self, notify=True, optional=False):
+        """ Create a new expression for observing items inside a dict.
+
+        Events emitted (if any) will be instances of
+        :py:class:`~traits.observers.events.DictChangeEvent`.
+
+        If an expression with ``dict_items`` is further extended, the
+        **values** of the dict will be given to the next item in the
+        expression. For example, the following observes a trait named "number"
+        on any object that is one of the values in the dict named "mapping"::
+
+            trait("mapping").dict_items().trait("number")
+
+        Parameters
+        ----------
+        notify : boolean, optional
+            Whether to notify for changes.
+        optional : booleal, optional
+            Whether to ignore this if the upstream object is not a dict.
+
+        Returns
+        -------
+        new_expression : traits.observers.expressions.Expression
+        """
+        return self.then(dict_items(notify=notify, optional=optional))
+
+    def list_items(self, notify=True, optional=False):
+        """ Create a new expression for observing items inside a list.
+
+        Events emitted (if any) will be instances of
+        :py:class:`~traits.observers.events.ListChangeEvent`.
+
+        e.g. ``trait("containers").list_items()`` for observing to mutations
```

Comment:
```suggestion
        e.g. ``trait("containers").list_items()`` for observing mutations
```

Reply:
Not sure if you missed this one, but it's still outstanding.

Reply:
Thanks!